core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -1652,9 +1652,15 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
 
     val scriptWitnessVec = getRecords(WitnessScriptKeyId)
     val scriptWitnessOpt =
-      if (scriptWitnessVec.size == 1)
+      if (scriptWitnessVec.size == 1) {
         Some(P2WSHWitnessV0(scriptWitnessVec.head.witnessScript))
-      else None
+      } else if (output.scriptPubKey
+                   .isInstanceOf[P2WPKHWitnessSPKV0] || redeemScriptOpt.exists(
+                   _.isInstanceOf[P2WPKHWitnessSPKV0])) {
+        Some(P2WPKHWitnessV0(signer.publicKey))
```

Comment:
as this could be a common pattern, it might be nice to add a `scriptWitnessOpt: Option[ScriptWitnessV0]` to `InputPSBTMap` so that we don't have to go digging for `P2WSH` and `P2WPKH` witnesses